### PR TITLE
EXILED 9 Optimized

### PR DESCRIPTION
**Pull request by ITeMbI4**
Refactor finished, please review

### CODE REVIEW

File: EXILED/Exiled.API/Features/Core/Generic/Pools/DictionaryPool.cs [hunk 1]
```diff
@@ -37,10 +37,17 @@ private DictionaryPool()
         /// <returns>The <see cref="Dictionary{TKey, TValue}"/>.</returns>
         public Dictionary<TKey, TValue> Get()
```
Comment (louis1706) [inline EXILED/Exiled.API/Features/Core/Generic/Pools/DictionaryPool.cs:38]: Need to be verified than it's work

**Review by louis1706 — CHANGES_REQUESTED** [review]
Need to check the pragma warning disable

File: EXILED/Exiled.API/Enums/EffectType.cs [hunk 2]
```diff
@@ -213,12 +213,6 @@ public enum EffectType
         /// </summary>
         SilentWalk,
 
-        /// <summary>
-        /// Makes you a marshmallow guy.
-        /// </summary>
-        [Obsolete("Not functional in-game")]
-        Marshmallow,
-
```
[the remark this answers is not in the document]
Comment (ITeMbI4) [reply EXILED/Exiled.API/Enums/EffectType.cs]: Fixed

File: EXILED/Exiled.API/Extensions/EffectTypeExtension.cs [hunk 3]
```diff
@@ -67,9 +68,6 @@ public static class EffectTypeExtension
             { EffectType.AntiScp207, typeof(AntiScp207) },
             { EffectType.Scanned, typeof(Scanned) },
             { EffectType.SilentWalk, typeof(SilentWalk) },
-#pragma warning disable CS0618
-            { EffectType.Marshmallow, typeof(MarshmallowEffect) },
-#pragma warning restore CS0618
```
[the remark this answers is not in the document]
Comment (ITeMbI4) [reply EXILED/Exiled.API/Extensions/EffectTypeExtension.cs]: Fixed

File: EXILED/Exiled.API/Extensions/DangerTypeExtensions.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ public static class DangerTypeExtensions
             { DangerType.PlayerDamaged, typeof(PlayerDamagedDanger) },
             { DangerType.ScpEncounter, typeof(ScpEncounterDanger) },
             { DangerType.ZombieEncounter, typeof(ZombieEncounterDanger) },
-            { DangerType.ArmedEnemy, typeof(ArmedEnemyDanger) },
+            { DangerType.ArmedEnemyEncounter, typeof(ArmedEnemyDanger) },
```
[the remark this answers is not in the document]
Comment (ITeMbI4) [reply EXILED/Exiled.API/Extensions/DangerTypeExtensions.cs]: Fixed

**Pull request closed without merging**